awriter/writer.go

```diff
@@ -126,6 +126,7 @@ func writeTempHeader(c artifact.Compressor, manifestChecksumStore *artifact.Chec
 	}()
 
 	if err != nil {
+		os.Remove(f.Name())
```

Comment:
Question: Will this not leak in the other error with exit paths also?

Reply:
All the other return paths are inside a lambda function. They return the very error that ends up here.